experiments/download_ssl4eo.py

```diff
@@ -393,98 +288,111 @@ def update(self, delta: int = 1) -> int:
             return self.value
 
 
-def fix_random_seeds(seed: int = 42) -> None:
-    """
-    Fix random seeds.
-    """
-    np.random.seed(seed)
-
-
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument("--save-path", type=str, default="./data/")  # dir to save data
-    parser.add_argument("--debug", action="store_true")
-    parser.add_argument("--cloud-pct", type=int, default=20)
-    parser.add_argument("--num-workers", type=int, default=8)
-    parser.add_argument("--log-freq", type=int, default=10)  # print frequency
     parser.add_argument(
-        "--resume", type=str, default=None
-    )  # resume from existing coordinates
+        "--save-path", type=str, default="./data/", help="dir to save data"
+    )
+    # collection properties
+    parser.add_argument(
+        "--collection", type=str, default="COPERNICUS/S2", help="GEE collection name"
+    )
+    parser.add_argument(
+        "--qa-band", type=str, default="QA60", help="qa band name (optional)"
+    )  # optional
+    parser.add_argument(
+        "--qa-cloud-bit", type=int, default=10, help="qa band cloud bit (optional)"
+    )  # optional
+    parser.add_argument(
+        "--meta-cloud-name",
+        type=str,
+        default="CLOUDY_PIXEL_PERCENTAGE",
+        help="meta data cloud percentage name",
+    )
+    parser.add_argument(
+        "--cloud-pct", type=int, default=20, help="cloud percentage threshold"
+    )
+    # patch properties
+    parser.add_argument(
+        "--dates",
+        type=str,
+        nargs="+",
+        default=["2021-12-21", "2021-09-22", "2021-06-21", "2021-03-20"],
+        help="reference dates",
+    )
+    parser.add_argument(
+        "--radius", type=int, default=1320, help="patch radius in meters"
+    )
+    parser.add_argument(
+        "--bands",
+        type=str,
+        nargs="+",
+        default=[
+            "B1",
+            "B2",
+            "B3",
+            "B4",
+            "B5",
+            "B6",
+            "B7",
+            "B8",
+            "B8A",
+            "B9",
+            "B10",
+            "B11",
+            "B12",
+        ],
+        help="bands to download",
+    )
+    parser.add_argument(
+        "--crops",
+        type=int,
+        nargs="+",
+        default=[44, 264, 264, 264, 132, 132, 132, 264, 132, 44, 44, 132, 132],
```

Comment:
Won't this always be equal to the radius? Or is this in case the radius is off by half a pixel or something? I wonder if we could combine these two args

Reply:
yes equal to  radius*2. the reason here is that different bands may have different resolutions which are not fixed...

Reply:
Is there a way to read the band resolutions from GEE? We're already passing the radius in geospatial coords to `get_patch`. Won't that automatically return patches with the right dimensions?

Reply:
Most patches will already return the right sizes, but some will return slightly different (e.g. 264x265). Thus the "crop" argument is also optional (can be None).